### ISSUE TITLE
symfony/phpunit-bridge equivalent for Behat

### ISSUE THREAD

**Issue by caciobanu** [Behat/Behat#883]
Working mainly on Symfony 2 projects we included https://github.com/symfony/phpunit-bridge to make sure we are not using any deprecated Symfony features. It works well with PHPUnit but we also do testing with Behat using the browser kit driver for mink so we lost that nice feature of seeing a list of deprecated features used. To fill the gap I've created a Behat extension that does just that: https://github.com/caciobanu/behat-deprecation-extension. I added as a dependency only behat/behat because the extension is usable within other non Symfony projects (I think the constraints are obvious).

If there is interest maybe we could move the extension to Behat organization repositories and so more people will find about it.

What do you guys think ?

**Comment by stof** [comment]
I will review your code as soon as I have time. Creating such a tool for Behat was on my TODO list since a long time, but I never started it.

**Comment by caciobanu** [comment]
I'm not sure that the way the extension is coded now is the right way because I don't know the internals of Behat to well. But the code can be changed and adapted to be more in line with Behat.

**Comment by stof** [comment]
which is exactly why I plan to review it :smile:

**Comment by carlos-granados** [comment]
Given that there is an extension which does its job pretty well (I have successfully used it in several projects in the past), I am not sure we really need to look into incorporating this into the core, so I would suggest to close this ticket. @acoulton @ttomdewit what do you think?

**Comment by ttomdewit** [comment]
> Given that there is an extension which does its job pretty well (I have successfully used it in several projects in the past), I am not sure we really need to look into incorporating this into the core, so I would suggest to close this ticket. @acoulton @ttomdewit what do you think?

I’d say closing is fine. Do we want to share that plugin somewhere? Do we endorse 3rd party as such? If it’s a well-working solution I’d not add it to core atm.

**Comment by acoulton** [comment]
Thanks for offering @caciobanu - I agree with @carlos-granados and @ttomdewit that we shouldn't adopt this into core. Mostly because we have limited resource and there's plenty for us to review & do on the existing official behat repos for now.

@ttomdewit we don't currently endorse / link to any third party extensions. I think that could be useful - I've created a docs issue https://github.com/Behat/docs/issues/150 for us to consider it.

I'll close this in the meantime.
